### ISSUE TITLE
Test build of master with MacOSX

### ISSUE THREAD

**Issue by jyoung3131** [hpcgarage/spatter#47]
Spatter can be compiled with MacOSX on Catalina using gcc and/or clang, but it requires some extra steps. Users need to install cmake and libomp via Homebrew and possibly also [set extra flags for OpenMP.](https://iscinumpy.gitlab.io/post/omp-on-high-sierra/). 

It would be nice to handle this in a better fashion by [checking to see if the platform is "APPLE"](https://cmake.org/cmake/help/v3.18/variable/APPLE.html?highlight=apple) and then either use FindOpenMP or set flags manually.

**Comment by jyoung3131** [comment]
This is addressed in branch "issue47" commit [822c4ae](https://github.com/hpcgarage/spatter/commit/822c4ae60533bbe32d6d3ef10351e4fee848bdb5) Added the CMake 3.10 way to check for and include OpenMP. 

Tests are still broken I think due to OpenMP relative paths for the omp.h header. 

```
cd /Users/jyoung/GitRepos/external/github/spatter/build_serial_gnu/tests && /usr/bin/gcc -DSG_DATA_TYPE=double -DUSE_POSIX_MEMALIGN -DUSE_SERIAL -I/Users/jyoung/GitRepos/external/github/spatter/build_serial_gnu/tests -I/Users/jyoung/GitRepos/external/github/spatter/tests -I/Users/jyoung/GitRepos/external/github/spatter/src/serial  -O3 -I/Users/jyoung/GitRepos/external/github/spatter/include -DSPAT_C="/usr/bin/gcc" -DSPAT_C_VER="11.0.3.11030032" -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk   -o CMakeFiles/parse_uniform_stride_1.dir/__/src/openmp/openmp_kernels.c.o   -c /Users/jyoung/GitRepos/external/github/spatter/src/openmp/openmp_kernels.c
In file included from /Users/jyoung/GitRepos/external/github/spatter/src/openmp/openmp_kernels.c:2:
/Users/jyoung/GitRepos/external/github/spatter/src/openmp/openmp_kernels.h:4:10: fatal error: 'omp.h' file not found
#include <omp.h>
```

**Comment by jyoung3131** [comment]
I believe what is happening is that the "add_directory" command for adding ctests breaks the include/linking of OpenMP. You can replicate this error by building the OpenMP GNU backend using the included configure script.

**Comment by jyoung3131** [comment]
Fixed in commit f4b27352d and tested with GNU and Clang on OSX Catalina.
